### DISCUSSION TITLE
Is there a way to delete old logs every time you run a program?

### DISCUSSION THREAD

**Discussion by wnsntgs** [gabime/spdlog#2144] [Q&A] [answered]
Let's suppose that I have `spdlog::basic_logger_mt("My Logger", "mylog.txt");` Now I manually check if a file with this name exists and manually delete it before initializing my logger. Actually, I don’t like it, but I haven’t found a way to automate it with a library. Is there a way to clear `mylog.txt` every time I run my program regardless of the current time or day of the week?

**Comment by tt4g** [comment, marked as answer]
If the `truncate` argument is set to `true`, the log file contents will be truncated on opening.
e.g. spdlog::basic_logger_mt("My Logger", "mylog.txt", true);

https://github.com/gabime/spdlog/blob/v1.9.2/include/spdlog/sinks/basic_file_sink.h#L43-L52
